examples/distributed-tracing/src/index.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+require 'vendor/autoload.php';
+
+use DI\Bridge\Slim\Bridge;
+use DI\ContainerBuilder;
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Promise\PromiseInterface;
+use GuzzleHttp\Promise\Utils;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\SDK\Trace\Propagation\TraceContextPropagator;
+use OpenTelemetry\SDK\Trace\Tracer;
+use OpenTelemetry\SDK\Trace\TracerProvider;
+use OpenTelemetry\SDK\Trace\TracerProviderFactory;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Server\RequestHandlerInterface as RequestHandler;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Slim\Routing\RouteContext;
+
+//create default tracer from environment variables, now available as TracerProvider::getDefaultTracer
+$tracer = (new TracerProviderFactory('example'))->create()->getTracer();
+
+$cb = new ContainerBuilder();
+$container = $cb->addDefinitions([
+    Tracer::class => $tracer,
+    Client::class => function() {
+        $stack = HandlerStack::create();
+        //a guzzle middleware to wrap http calls in a span, and inject trace headers
+        $stack->push(function(callable $handler) {
+            return function (RequestInterface $request, array $options) use ($handler): PromiseInterface {
+                $span = TracerProvider::getDefaultTracer()
+                    ->spanBuilder(sprintf('%s %s', $request->getMethod(), $request->getUri()))
+                    ->setSpanKind(SpanKind::KIND_CLIENT)
+                    ->setAttribute('http.method', $request->getMethod())
+                    ->setAttribute('http.url', $request->getUri())
+                    ->startSpan();
+                $ctx = $span->storeInContext(Context::getCurrent());
```

Comment:
This works, but it looks like a few revisions ago you could just pass a SpanContext for trace propagation, which would be simpler since a SpanContext is what it ultimately works with...